clients/src/test/java/org/apache/kafka/common/requests/OffsetFetchResponseTest.java

```diff
@@ -159,14 +159,14 @@ public void testShouldThrottle() {
 
     @Test
     public void testNullableMetadata() {
+        PartitionData pd = new PartitionData(
+            offset,
+            leaderEpochOne,
+            null,
+            Errors.UNKNOWN_TOPIC_OR_PARTITION);
+        assertEquals(pd, pd);
```

Comment:
I believe that one could be easily tempted to remove such assertion without knowing the context. I would add a comment to explain it or define a dedicate test case for the bug. I am fine with either ways.

Reply:
thanks for this quickly and meaningful review and I will copy that :)